tsdat/io/storage.py

```diff
@@ -90,13 +97,37 @@ def save_data(self, dataset: xr.Dataset):
             dataset (xr.Dataset): The dataset to save.
 
         -----------------------------------------------------------------------------"""
-        datastream = dataset.attrs["datastream"]
-        filepath = self._get_dataset_filepath(dataset, datastream)
-        filepath.parent.mkdir(exist_ok=True, parents=True)
-        self.handler.writer.write(dataset, filepath)
-        logger.info("Saved %s dataset to %s", datastream, filepath.as_posix())
 
-    def fetch_data(self, start: datetime, end: datetime, datastream: str) -> xr.Dataset:
+        data_filepath = self._get_dataset_filepath(dataset, self.parameters.by_date)
+
+        data_filepath.parent.mkdir(exist_ok=True, parents=True)
+        self.handler.writer.write(dataset, data_filepath)
+        logger.info(
+            "Saved %s dataset to %s", dataset.datastream, data_filepath.as_posix()
+        )
+
+    def save_ancillary_file(self, temp_filepath: Path, dataset: xr.Dataset):
```

Comment:
Arguments should stay the same as before -- this function shouldn't require any changes